daft/filesystem.py

```diff
@@ -214,58 +196,89 @@ def _resolve_path_and_filesystem(
-    from fsspec.implementations.http import HTTPFileSystem
+    protocol = get_protocol_from_path(path)
+    translated_kwargs: dict[str, Any]
+
+    def _set_if_not_none(kwargs: dict[str, Any], key: str, val: Any | None):
+        """Helper method used when setting kwargs for pyarrow"""
+        if val is not None:
+            kwargs[key] = val
+
+    ###
+    # S3
+    ###
+    if protocol == "s3":
+        translated_kwargs = {}
+        if io_config is not None and io_config.s3 is not None:
+            s3_config = io_config.s3
+            _set_if_not_none(translated_kwargs, "endpoint_override", s3_config.endpoint_url)
+            _set_if_not_none(translated_kwargs, "access_key", s3_config.key_id)
+            _set_if_not_none(translated_kwargs, "secret_key", s3_config.access_key)
+            _set_if_not_none(translated_kwargs, "session_token", s3_config.session_token)
+            _set_if_not_none(translated_kwargs, "region", s3_config.region_name)
+            _set_if_not_none(translated_kwargs, "anonymous", s3_config.anonymous)
+
+        resolved_filesystem = S3FileSystem(**translated_kwargs)
+        resolved_path = resolved_filesystem.normalize_path(_unwrap_protocol(path))
+        return resolved_path, resolved_filesystem
+
+    ###
+    # Local
+    ###
+    elif protocol == "file":
+        resolved_filesystem = LocalFileSystem()
+        resolved_path = resolved_filesystem.normalize_path(_unwrap_protocol(path))
+        return resolved_path, resolved_filesystem
+
+    ###
+    # GCS
+    ###
+    elif protocol in {"gs", "gcs"}:
+        try:
+            from pyarrow.fs import GcsFileSystem
+        except ImportError:
+            raise ImportError(
+                "Unable to import GcsFileSystem from pyarrow - please ensure you have pyarrow >= 9.0 or consider "
+                "using Daft's native GCS IO code instead"
+            )
+
+        translated_kwargs = {}
+        if io_config is not None and io_config.gcs is not None:
+            gcs_config = io_config.gcs
+            _set_if_not_none(translated_kwargs, "anonymous", gcs_config.anonymous)
+            _set_if_not_none(translated_kwargs, "project_id", gcs_config.project_id)
+
+        resolved_filesystem = GcsFileSystem(**translated_kwargs)
+        resolved_path = resolved_filesystem.normalize_path(_unwrap_protocol(path))
+        return resolved_path, resolved_filesystem
+
+    ###
+    # HTTP: Use FSSpec as a fallback
+    ###
+    elif protocol in {"http", "https"}:
+        fsspec_fs_cls = get_filesystem_class(protocol)
+        fsspec_fs = fsspec_fs_cls()
+        resolved_filesystem, resolved_path = pafs_resolve_filesystem_and_path(path, fsspec_fs)
+        resolved_path = resolved_filesystem.normalize_path(resolved_path)
+        return resolved_path, resolved_filesystem
+
+    ###
+    # Azure: Unsupported! Users should use our native IO layer if possible.
+    ###
+    elif protocol in {"az", "abfs"}:
```

Comment:
couldnt this return the [fsspec azure filesystem](https://github.com/fsspec/adlfs).

Reply:
Oh yes! Changed, added and tested manually.